fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -493,4 +433,164 @@ def loop_over_jacobians(controller, options, grabbed_output):
+                chi_sq = controller.eval_chisq(
+                    params=controller.final_params,
+                    x=controller.data_x,
+                    y=controller.data_y,
+                    e=controller.data_e)
+
+                chi_sq_check = any(np.isnan(n) for n in chi_sq) \
+                    if problem.multifit else np.isnan(chi_sq)
+                if np.isnan(runtime) or chi_sq_check:
+                    raise ControllerAttributeError(
+                        "Either the computed runtime or chi_sq values "
+                        "was a NaN.")
+            # Catching all exceptions as this means runtime cannot be
+            # calculated
+            # pylint: disable=broad-except
+            except Exception as excp:
+                LOGGER.warning(str(excp))
+
+                runtime = np.inf
+                controller.flag = 3
+                controller.final_params = \
+                    None if not problem.multifit \
+                    else [None] * len(controller.data_x)
+
+                chi_sq = np.inf if not problem.multifit \
+                    else [np.inf] * len(controller.data_x)
+
+            # If bounds have been set, check that they have
+            # been respected by the minimizer and set error
+            # flag if not
+            if controller.problem.value_ranges is not None \
+                    and controller.flag != 3:
+                controller.check_bounds_respected()
+
+            # record algorithm type for specified minimizer
+            type_str = controller.record_alg_type(
+                minimizer, options.algorithm_type)
+            options.minimizer_alg_type[minimizer_name] = type_str
+
+            result_args = {'options': options,
+                           'cost_func': cost_func,
+                           'jac': jacobian,
+                           'chi_sq': chi_sq,
+                           'runtime': runtime,
+                           'minimizer': minimizer_name,
+                           'initial_params': controller.initial_params,
+                           'params': controller.final_params,
+                           'error_flag': controller.flag,
+                           'name': problem.name}
+            new_result.append(result_args)
+            new_minimizer_list.append(minimizer_name)
+            if problem.multifit:
+                # for multifit problems, multiple chi_sq values are stored
+                # in a list, which should extend the empty new_chi_sq list
+                # rather than be appended to it
+                new_chi_sq.extend(chi_sq)
+            else:
+                new_chi_sq.append(chi_sq)
+
+            # For minimizers that do not accept hessians we raise an
+            # StopIteration exception to exit the loop through the
+            # Hessians
+            if not minimizer_check:
+                raise StopIteration
+    except StopIteration:
+        pass
```

Comment:
This is equivalent to using `break` but more complicated. The Jacobian implementation is slightly different as it has to escape 2 levels of for loops